scripts/gen_launch.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+
+# USAGE: after changing tests, generate a launch.json configuration with this script
+# ./scripts/gen_launch.py > .vscode/launch.json
```

Comment:
Just double-check, I tried this without any test change, and it overwrote the original `launch.json` with an empty file. Is this expected?

Reply:
This is not expected and means the script failed. Was anything printed to STDOUT/STDERR?

Reply:
I first got this error:
```
[nix-shell:~/code/hotshot]$ ./scripts/gen_launch.py > .vscode/launch.json
RUST_LOG=none cargo test --verbose --profile=release-lto --features=full-ci,channel-async-std --lib --bins --tests --benches --package=benchmark_client --no-fail-fast -- --test-threads=1 -Zunstable-options --format json
error: no library targets found in package `benchmark_client`
error: Recipe `list_tests_json` failed on line 37 with exit code 101
RUST_LOG=none cargo test --verbose --profile=release-lto --features=full-ci,channel-async-std --lib --bins --tests --benches --package=hotshot --no-fail-fast -- --test-threads=1 -Zunstable-options --format json
```
It continued on more compilations, then failed with:
```
error: Failed to delete invalidated or incompatible incremental compilation session directory contents `/Users/keyao/Code/hotshot/target_dirs/nix_rustc/release-lto/incremental/libp2p-1blzcdq7m79yy/s-gio394f2s8-aim7c6-working/dep-graph.bin`: No such file or directory (os error 2).

error: failed to create dependency graph at `/Users/keyao/Code/hotshot/target_dirs/nix_rustc/release-lto/incremental/libp2p-1blzcdq7m79yy/s-gio394f2s8-aim7c6-working/dep-graph.part.bin`: No such file or directory (os error 2)
```
And `launch.json` became empty.

I'll try deleting `target_dirs` since the latter two errors are related to it.

Reply:
> I first got this error:
> 
> ```
> [nix-shell:~/code/hotshot]$ ./scripts/gen_launch.py > .vscode/launch.json
> RUST_LOG=none cargo test --verbose --profile=release-lto --features=full-ci,channel-async-std --lib --bins --tests --benches --package=benchmark_client --no-fail-fast -- --test-threads=1 -Zunstable-options --format json
> error: no library targets found in package `benchmark_client`
> error: Recipe `list_tests_json` failed on line 37 with exit code 101
> RUST_LOG=none cargo test --verbose --profile=release-lto --features=full-ci,channel-async-std --lib --bins --tests --benches --package=hotshot --no-fail-fast -- --test-threads=1 -Zunstable-options --format json
> ```

After deleting `target_dirs`,  there's no error related to `target_dirs`, but I still got this error above and the "linking with cc failed" error.

Reply:
Oh, actually, though there were the above-mentioned errors, the compilation finished at the end, and `launch.json` was restored. There were some changes though; lines 12-131 and 152-631 were deleted.

Reply:
The first error (about benchmark client) is expected and shouldn't change anything. Those other errors are weird..

Reply:
I'm not sure what to make of the deleted lines... Maybe only one package successfully finished? Are there only centralized web sever crate tests available?

Reply:
Turned out that anything with the `hotshot`, `hotshot-testing`, or `hotshot-centralized-web-server` package was deleted, and I suspect it's due to the "linking with cc" errors.

And yes, `just list_tests_json libp2p-networking` works.

Reply:
Ah, gotcha. Do `just list_test_json hotshot`,  `just list_test_json hotshot-testing`, and `hotshot-centralized-web-server` all run successfully starting from a state where `target_dirs` is empty? If so I can add in a `rm -rf target_dirs` or `cargo clean` invocation between runs.

I wonder if the incremental builds are confusing cargo.